src/ekam/rules/test.ekam-rule

```diff
@@ -55,10 +55,12 @@ fi
 echo newOutput "${1}.log"
 read TEST_LOG
 
-if LD_PRELOAD=$INTERCEPTOR DYLD_FORCE_FLAT_NAMESPACE= DYLD_INSERT_LIBRARIES=$INTERCEPTOR \
-   $INTERPRETER $TEST_PROG 3>&1 4<&0 2>"$TEST_LOG" >&2; then
-	echo passed
-	exit 0
+TIMEOUT_CMD="timeout -s 9 ${TEST_TIMEOUT:-30m}"
+
+if $TIMEOUT_CMD env LD_PRELOAD=$INTERCEPTOR DYLD_FORCE_FLAT_NAMESPACE= DYLD_INSERT_LIBRARIES=$INTERCEPTOR \
```

Comment:
@kentonv I tried putting the timeout after the LD_PRELOAD & stripping the tmp/ prefix but it fails to run on ARM64 for some reason. This method seems to actually work & I think has no ill side-effects to worry about.